### PR TITLE
[PDI-19098] Variable name needs to be compared to the incoming row

### PR DESCRIPTION
**Pull request by peterrinehart**
fields in a case-insensitive manner in order to match previous
functionality.

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 32m 5s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
:ok_hand: All tests passed!
 
Tests run: **3926**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-8020/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: engine/src/main/java/org/pentaho/di/trans/steps/transexecutor/TransExecutor.java [hunk 1]
```diff
@@ -374,6 +374,34 @@ void passParametersToTrans( List<String> incomingFieldValues ) throws KettleExce
...
+   * @param value String to search for
+   * @return true if the value is found, false otherwise
+   */
+  private boolean listContainsIgnoreCase( final List<String> list, final String value ) {
+    return list.stream().anyMatch( s -> s.equalsIgnoreCase( value ) );
+  }
+
+  /**
+   * Get the index of the specified value in the list, ignoring case.  Should only be called
+   * after verifying that the value exists first using listContainsIgnoreCase!
```
Comment (ricksjames) [inline engine/src/main/java/org/pentaho/di/trans/steps/transexecutor/TransExecutor.java:389]: Don't think this comment is necessary, ArrayList.indexOf() also returns -1 if not in the list

File: engine/src/main/java/org/pentaho/di/trans/steps/transexecutor/TransExecutor.java [hunk 2]
```diff
@@ -374,6 +374,34 @@ void passParametersToTrans( List<String> incomingFieldValues ) throws KettleExce
...
+   * @param value String to search for
+   * @return the index of the value in the list if found, -1 if not found
+   */
+  private int getIndexIgnoreCase( final List<String> list, final String value ) {
+    for ( int i = 0; i < list.size(); i++ ) {
+      if ( list.get( i ).equalsIgnoreCase( value ) ) {
+        return i;
+      }
+    }
+    // THIS SHOULD NEVER HAPPEN BECAUSE WE SHOULD NEVER CALL THIS METHOD WITHOUT FIRST CHECKING FOR THE EXISTENCE OF
```
Comment (ricksjames) [inline engine/src/main/java/org/pentaho/di/trans/steps/transexecutor/TransExecutor.java:400]: Don't think this comment is necessary, ArrayList.indexOf() also returns -1 if not in the list

File: engine/src/test/java/org/pentaho/di/trans/steps/transexecutor/TransExecutorUnitTest.java [hunk 3]
```diff
@@ -498,6 +498,49 @@ public void testExecuteTransWithFieldsAndNoInput() throws KettleException {
...
+    executor.setVariable( paramOverwrite, parentValue );
+    executor.setVariable( childParam, childValue );
+
+    RowMetaInterface inputRowMeta = mock( RowMetaInterface.class );
+
+    Mockito.when( executor.getLogLevel() ).thenReturn( LogLevel.NOTHING );
+    parent.setLog( new LogChannel( this ) );
+    Mockito.doCallRealMethod().when( executor ).createInternalTrans( );
+    Mockito.when(  executor.getData().getExecutorTransMeta().listVariables() ).thenReturn( new String[0] );
+    Mockito.when(  executor.getData().getExecutorTransMeta().listParameters() ).thenReturn( new String[0] /*{parentParam}*/ );
```
Comment (ricksjames) [inline engine/src/test/java/org/pentaho/di/trans/steps/transexecutor/TransExecutorUnitTest.java:528]: did you mean to leave in this commented out code?

File: engine/src/main/java/org/pentaho/di/trans/steps/transexecutor/TransExecutor.java (hunk 2 above)
Comment (peterrinehart) [reply engine/src/main/java/org/pentaho/di/trans/steps/transexecutor/TransExecutor.java:400]: I just want to be certain that the intent of the code is clear when someone else has to maintain it next.  If we wind up in the situation where this method has to return -1 (i.e. the current assumptions are broken), the current exception handling won't make it very clear what the actual problem is

File: engine/src/main/java/org/pentaho/di/trans/steps/transexecutor/TransExecutor.java (hunk 1 above)
Comment (peterrinehart) [reply engine/src/main/java/org/pentaho/di/trans/steps/transexecutor/TransExecutor.java:389]: See below.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 33m 27s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
:ok_hand: All tests passed!
 
Tests run: **3926**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-8020/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by ricksjames — APPROVED** [review]

**Pull request merged**